src/main/java/software/sigma/bu003/internship/coparts/service/PartService.java

```diff
@@ -44,7 +47,12 @@ public void deletePart(String brand, String code) {
     }
 
     private Part checkIfPartExistsInDB(String brand, String code){
-        return partRepository.findById(brand + code)
+        return partRepository.findById(brand.replaceAll("\\s","") + code)
```

Comment:
Because some brands have a name like "BMW Parts" with a space in the middle. In order to follow the link and find such a part, I remove the extra punctuation